mobile/lib/features/wallet/share_invoice_screen.dart

```diff
@@ -0,0 +1,139 @@
+                      version: QrVersions.auto,
+                      size: 200.0,
+                    ),
+                  ),
+                ),
+              ]),
+            ),
+            Row(children: [
+              Flexible(
+                flex: 1,
+                child: Padding(
+                  padding: buttonSpacing,
+                  child: OutlinedButton(
+                    onPressed: () {
+                      Clipboard.setData(ClipboardData(text: invoice)).then((_) {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                            const SnackBar(content: Text('Invoice copied to clipboard')));
+                      });
+                    },
+                    style: ElevatedButton.styleFrom(
+                      shape: const RoundedRectangleBorder(
+                          borderRadius: BorderRadius.all(Radius.circular(5.0))),
+                    ),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: const [
+                        Padding(
+                          padding: EdgeInsets.symmetric(horizontal: 8.0),
+                          child: Icon(Icons.copy),
+                        ),
+                        Text("Copy"),
+                      ],
+                    ),
+                  ),
+                ),
+              ),
+              Flexible(
+                flex: 1,
+                child: Padding(
+                  padding: buttonSpacing,
+                  child: OutlinedButton(
+                    onPressed: () => Share.share(invoice),
+                    style: ElevatedButton.styleFrom(
+                      shape: const RoundedRectangleBorder(
+                          borderRadius: BorderRadius.all(Radius.circular(5.0))),
+                    ),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: const [
+                        Padding(
+                          padding: EdgeInsets.symmetric(horizontal: 8.0),
+                          child: Icon(Icons.send),
+                        ),
+                        Text("Share"),
+                      ],
+                    ),
+                  ),
+                ),
+              )
+            ]),
+            const Padding(
+              padding: EdgeInsets.all(8.0),
+              child: Center(
+                child: Text(
+                  "You will see the incoming payment once you send the funds.",
+                  style: TextStyle(color: Colors.grey),
+                ),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
+              child: ElevatedButton(
+                onPressed: () {
+                  // Pop both create invoice screen and share invoice screen
+                  GoRouter.of(context).pop();
```

Comment:
I feel like this does not quite capture what "Done" semantically embodies - that flow of actions is finished and we are returning to where we were before. I believe that if we just use `GoRouter.of(context).go(WalletScreen.route)` and then clicked a hardware back button, this would take us back to the share invoice screen again, no? This seems wrong to me

Reply:
> I feel like this does not quite capture what "Done" semantically embodies - that flow of actions is finished and we are returning to where we were before. I believe that if we just use `GoRouter.of(context).go(WalletScreen.route)` and then clicked a hardware back button, this would take us back to the share invoice screen again, no? This seems wrong to me.

Great point, I haven't thought about that 🙂

> Does popping twice introduce some kind of race condition?
I guess not? However, it is a bit failure prune as if we are going to add another screen to the flow, it has to be ensured that we will add another pop to the Done Button. I am wondering if there isn't a better way to get back to where to flow started without running into the back button issue.

Reply:
> > Does popping twice introduce some kind of race condition?
> 
> I guess not? However, it is a bit failure prune as if we are going to add another screen to the flow, it has to be ensured that we will add another pop to the Done Button. I am wondering if there isn't a better way to get back to where to flow started without running into the back button issue.

Looks like what we would want here is `pop_until` , but that is currently not supported in `go_router` as far as I could see. I found this draft on the matter: https://github.com/flutter/packages/pull/2728

Note: I think for now popping twice is fine. We could add a test that steps through the flow to ensure that we are at a certain route (location) after finishing. That way adding an additional screen would be covered by the test.